adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -162,6 +165,232 @@ func (b *BM25Searcher) retrieveScoreAndSortForSingleTerm(ctx context.Context,
+
+	// If there is only one, we are finished
+	if len(idLists) == 1 {
+		return idLists[0]
+	}
+
+	docHash := make(map[uint64]docPointerWithScore)
+
+	total := 0
+
+	for i, list := range idLists {
+		for _, doc := range list.docIDs {
+			// if id is not in the map, add it
+			if _, ok := docHash[doc.id]; !ok {
+				if doc.Additional == nil {
+					doc.Additional = make(map[string]interface{})
+				}
+				doc.Additional["BM25F_"+propNames[i]+"_frequency"] = fmt.Sprintf("%v", doc.frequency)
+				doc.Additional["BM25F_"+propNames[i]+"_propLength"] = fmt.Sprintf("%v", doc.propLength)
+				docHash[doc.id] = doc
+			} else {
+				// if id is in the map, add the frequency
+				existing := docHash[doc.id]
+				existing.frequency += doc.frequency
+				existing.propLength += doc.propLength
+				existing.Additional["BM25F_"+propNames[i]+"_frequency"] = fmt.Sprintf("%v", doc.frequency)
+				existing.Additional["BM25F_"+propNames[i]+"_propLength"] = fmt.Sprintf("%v", doc.propLength)
+				// TODO: We will have a different propLength for each property, how do we combine them?
+				docHash[doc.id] = existing
+			}
+		}
+		total += int(list.count)
+	}
+
+	// Make list from docHash
+	res := docPointersWithScore{
+		docIDs: make([]docPointerWithScore, len(docHash)),
+		count:  uint64(total),
+	}
+
+	i := 0
+	for _, v := range docHash {
+		res.docIDs[i] = v
+		i++
+	}
+
+
+
+	//Iterate over the list, get the propLength for all searched properties, add them together to find the docLength
+	for i, docId := range res.docIDs {
+		
+			doc := objectByIndexID(docId.id)
+			if doc== nil {
+				
+				continue
+			}
+
+		
+		if doc !=nil {
+			//Unpack each property and add the propLength to the docLength
+				propLength := 0
+				for _, propName := range propNames {
+					prop := doc.Object.Properties.(map[string]interface{})[propName]
+					propStr := fmt.Sprintf("%v", prop)
+					words := strings.Fields(propStr)  //FIXME use proper splitter
+						propLength +=len(words)
+					
+				}
+				if res.docIDs[i].Additional == nil {
+					res.docIDs[i].Additional = make(map[string]interface{})
+				}
+				res.docIDs[i].Additional["BM25F_docLength"] = fmt.Sprintf("%v", propLength)
+				
+			}
+			
+		}
+
+	
+
+
+	return res
+}
+
+// BM25F search each given property for a single term.  Results will be combined later
+func (b *BM25Searcher) retrieveForSingleTermMultipleProps(ctx context.Context, objectByIndexID func(index uint64)*storobj.Object,
+	properties []string, term string,
+) (docPointersWithScore, error) {
+	idss := []docPointersWithScore{}
+
+	searchTerm := term
+
+	propNames := []string{}
+
+	for _, propertyWithBoost := range properties {
+		boost := 1
+		var property = propertyWithBoost
+		if strings.Contains(propertyWithBoost, "^") {
+			property = strings.Split(propertyWithBoost, "^")[0]
+			boostStr := strings.Split(propertyWithBoost, "^")[1]
+			boost, _ = strconv.Atoi(boostStr)
+			fmt.Printf("Boosting %s by %d\n", property, boost)
+		}
```

Comment:
is there any validation for boost expression? like multiple `^`? float/negative/non-number values?
would be nice to have some tests for that